test/testnative.c

```diff
@@ -189,15 +189,15 @@ int main(int argc, char *argv[])
         quit(2);
     }
     for (i = 0; i < NUM_SPRITES; ++i) {
-        positions[i].x = (float)(SDL_rand_n(window_w - (int)sprite_w));
-        positions[i].y = (float)(SDL_rand_n(window_h - (int)sprite_h));
+        positions[i].x = (float)(SDL_rand(window_w - (int)sprite_w));
+        positions[i].y = (float)(SDL_rand(window_h - (int)sprite_h));
         positions[i].w = sprite_w;
         positions[i].h = sprite_h;
         velocities[i].x = 0.0f;
         velocities[i].y = 0.0f;
         while (velocities[i].x == 0.f && velocities[i].y == 0.f) {
-            velocities[i].x = (float)(SDL_rand_n(MAX_SPEED * 2 + 1) - MAX_SPEED);
-            velocities[i].y = (float)(SDL_rand_n(MAX_SPEED * 2 + 1) - MAX_SPEED);
+            velocities[i].x = (float)(SDL_rand(MAX_SPEED * 2 + 1) - MAX_SPEED);
+            velocities[i].y = (float)(SDL_rand(MAX_SPEED * 2 + 1) - MAX_SPEED);
```

Comment:
If sprite_w is float, maybe this is a good candidate to switch to SDL_randf

Reply:
These sprite bouncing tests use integer pixels, not subpixel velocities.

Reply:
I selected too many lines. The cast to int on lines 192, 193 made me think sprint_w,h are float.

Reply:
Oh, they are, but they're still integral pixels.